alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/UIConfig.java

```diff
@@ -57,7 +59,7 @@ public void setConfigFields() {
     protected abstract List<ConfigField> createFields();
 
     public List<ConfigField> getFields() {
-        return configFields;
+        return configFields.stream().map(SerializationUtils::clone).collect(Collectors.toList());
```

Comment:
For User's that have READONLY access we were changing the state of the fields. This change was being persisted even when you switched user's. So now we create a copy of the list to prevent the changes from effecting the in memory fields.